### PR TITLE
[Checkout] Show and choose available shipping methods

### LINKED ISSUE #21: As a Customer, I want to get available shipping methods with estimated costs
* Get list of shipping methods with estimated costs

```
GET /cart/{token}/available-shipping-methods
```

### PR DESCRIPTION
**Pull request by lchrusciel**
Resolves #21 and #19

Based on #64 

There is a slight change in a choosing request. Instead of `PUT /cart/{token}/select-shipping` with the following body: 
```json
{
    "shipments": [
        {
            "method": "DELIVERY_CODE"
        }
    ]
}
```
I have decide for `PUT /cart/{token}/shipping/{shippingId}` with
```json
{
    "method": "DELIVERY_CODE"
}
```

### CODE REVIEW

File: src/Controller/Checkout/ShowAvailableShippingMethodsAction.php [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+    private function getCalculatedShippingMethods(ShipmentInterface $shipment, $locale)
+    {
+        $rawShippingMethods = [];
+
+        /** @var ShippingMethodInterface $shippingMethod */
+        foreach ($this->shippingMethodsResolver->getSupportedMethods($shipment) as $shippingMethod) {
+            /** @var CalculatorInterface $calculator */
+            $calculator = $this->calculators->get($shippingMethod->getCalculator());
+
+            $shippingMethodView = new ShipmentMethodView();
```
Comment (lchrusciel) [inline src/Controller/Checkout/ShowAvailableShippingMethodsAction.php]: This logic should be extracted to custom factory

Comment (lchrusciel) [inline src/Controller/Checkout/ShowAvailableShippingMethodsAction.php]: Will be done in separate PR

File: src/Command/ChooseShippingMethod.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    private $shippingMethod;
+
+    /**
+     * @var string
+     */
+    private $orderToken;
+
+    /**
+     * @param string $orderToken
+     * @param mixed$shippingIdentifier
```
Comment (lchrusciel) [inline src/Command/ChooseShippingMethod.php]: missing space between `mixed$shippingIdentifier`

File: src/Command/ChooseShippingMethod.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace Sylius\ShopApiPlugin\Command;
+
+final class ChooseShippingMethod
+{
+    /**
+     * @var mixed
+     */
+    private $shippingIdentifier;
```
Comment (lchrusciel) [inline src/Command/ChooseShippingMethod.php]: `shipping` -> `shipment`

File: src/Controller/Checkout/ShowAvailableShippingMethodsAction.php [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+     */
+    private $cartRepository;
+
+    /**
+     * @var ViewHandlerInterface
+     */
+    private $viewHandler;
+
+    /**
+     * @var ViewHandlerInterface
```
Comment (lchrusciel) [inline src/Controller/Checkout/ShowAvailableShippingMethodsAction.php]: Wrong typehint

File: src/Handler/PutSimpleItemToCartHandler.php [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+        $this->productRepository = $productRepository;
+        $this->cartItemFactory = $cartItemFactory;
+        $this->orderItemModifier = $orderItemModifier;
+        $this->orderProcessor = $orderProcessor;
+        $this->manager = $manager;
+    }
+
+    public function handle(PutSimpleItemToCart $putSimpleItemToCart)
+    {
+        $cart = $this->cartRepository->findOneBy(['tokenValue' => $putSimpleItemToCart->token()]);
```
Comment (lchrusciel) [inline src/Handler/PutSimpleItemToCartHandler.php]: Missing check if cart is not empty

Comment (lchrusciel) [inline src/Handler/PutSimpleItemToCartHandler.php]: Done.

**Comment by pjedrzejewski** [comment]
Nice work Łukasz!
